repository/channel_impl_test.go

```diff
@@ -148,15 +148,54 @@ func TestGormRepository_GetChannelStats(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
 		t.Parallel()
 
+		channel := mustMakeChannel(t, repo, rand)
+		user1 := mustMakeUser(t, repo, rand)
+		user2 := mustMakeUser(t, repo, rand)
+		stamp1 := mustMakeStamp(t, repo, rand, user1.GetID())
+		stamp2 := mustMakeStamp(t, repo, rand, user1.GetID())
+
+		var u1Message, u2Message []*model.Message
+		u1Message = make([]*model.Message, 13)
+		u2Message = make([]*model.Message, 14)
+
+		for i := 0; i < 13; i++ {
+			u1Message[i] = mustMakeMessage(t, repo, user1.GetID(), channel.ID)
+		}
+
 		for i := 0; i < 14; i++ {
-			mustMakeMessage(t, repo, user.GetID(), channel.ID)
+			u2Message[i] = mustMakeMessage(t, repo, user2.GetID(), channel.ID)
+		}
+		require.NoError(t, repo.DeleteMessage(u2Message[12].ID))
+		require.NoError(t, repo.DeleteMessage(u2Message[13].ID))
+
+		for i := 0; i < 7; i++ {
+			mustAddMessageStamp(t, repo, u1Message[i].ID, stamp1.ID, user1.GetID())
+			mustAddMessageStamp(t, repo, u1Message[i].ID, stamp1.ID, user2.GetID())
+		}
+
+		for i := 0; i < 12; i++ {
+			mustAddMessageStamp(t, repo, u2Message[i].ID, stamp2.ID, user1.GetID())
+			mustAddMessageStamp(t, repo, u2Message[i].ID, stamp2.ID, user1.GetID())
 		}
-		require.NoError(t, repo.DeleteMessage(mustMakeMessage(t, repo, user.GetID(), channel.ID).ID))
 
 		stats, err := repo.GetChannelStats(channel.ID)
 		if assert.NoError(t, err) {
 			assert.NotEmpty(t, stats.DateTime)
-			assert.EqualValues(t, 15, stats.TotalMessageCount)
+
+			assert.EqualValues(t, 27, stats.TotalMessageCount)
+
+			assert.EqualValues(t, user2.GetID(), stats.Users[0].ID)
+			assert.EqualValues(t, 14, stats.Users[0].MessageCount)
+			assert.EqualValues(t, user1.GetID(), stats.Users[1].ID)
+			assert.EqualValues(t, 13, stats.Users[1].MessageCount)
```

Comment:
ここの前に`if assert.Len(t, stats.Users, 2)`とあると良さそうです
```suggestion

			if assert.Len(t, stats.Users, 2) {
				assert.EqualValues(t, user2.GetID(), stats.Users[0].ID)
				assert.EqualValues(t, 14, stats.Users[0].MessageCount)
				assert.EqualValues(t, user1.GetID(), stats.Users[1].ID)
				assert.EqualValues(t, 13, stats.Users[1].MessageCount)
			}
```